src/main/resources/db/migration/V5.3.0.3__7551-expanded-compound-datasetfield-validation.sql

```diff
@@ -0,0 +1,16 @@
+-- update required attribute of parents of compound fields where a subfield is required
+UPDATE datasetfieldtype SET required=true 
+WHERE id IN 
+(SELECT parentdatasetfieldtype_id FROM datasetfieldtype WHERE required=true);
+
+
+-- add input level rows for parents of compound fields  where a subfield has required input level
+-- (first run a delete to clear out any possible existing parent values)
+DELETE FROM dataversefieldtypeinputlevel WHERE datasetfieldtype_id in (
+SELECT parentdatasetfieldtype_id FROM datasetfieldtype dsft);
+  
+INSERT INTO dataversefieldtypeinputlevel (include,required,datasetfieldtype_id,dataverse_id)
+SELECT true, true, parentdatasetfieldtype_id, dataverse_id FROM
+dataversefieldtypeinputlevel dsftil, datasetfieldtype dsft
+WHERE dsftil.datasetfieldtype_id = dsft.id
+AND parentdatasetfieldtype_id IS NOT null;
```

Comment:
This does not modify the Producer Name to make it conditionally required to match what is happening in citation.tsv 

We need to add something like:

update datasetfieldtype set required = true where name = 'producerName';

Reply:
Right. I initially wrote these script to maintain the "status quo". But as producerName is going to conditionally required we either need this here or to change the release notes to  require reloading the tsv file.

Reply:
I wasn't sure about the intent of making the update correspond to the tsv, so whichever makes the most sense.

Reply:
OK, I decided to have them reload citation tsv to get the Producer Name change. (in this way we make sure they get the latest citation.tsv in case they decide to make any other changes).